etl/steps/data/meadow/countries/2023-09-25/isd.py

```diff
@@ -0,0 +1,43 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import numpy as np
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("isd.xlsx")
+
+    # Load data from snapshot.
+    tb = snap.read()
+
+    #
+    # Process data.
+    #
+    # Ensure all columns are snake-case, set an appropriate index, and sort conveniently.
+    tb = tb.underscore().set_index(["cownum", "start", "end"], verify_integrity=True).sort_index()
+
+    # Dtypes
+    tb = tb.astype({"othername": str, "decdate": str, "latitude": str, "longitude": str})
+
+    # Replace -9 -> NaN
+    tb = tb.replace(-9, np.nan)
+    # Remove -9 from categorical columns
+    columns_cat = tb.dtypes[tb.dtypes.isin(["category", "object"])].index
+    tb[columns_cat] = tb[columns_cat].astype(str).replace("-9", np.nan).astype("category")
```

Comment:
This is because values for categories are kept even if the `tb = tb.replace(-9, np.nan)` operation is in place, right?

Reply:
Yes, but I've just simplified this. Thanks for raising this!